server/src-lib/Hasura/RQL/DDL/RemoteSchema.hs

```diff
@@ -85,9 +85,14 @@ runReloadRemoteSchema
   :: (QErrM m, CacheRWM m)
   => RemoteSchemaNameQuery -> m EncJSON
 runReloadRemoteSchema (RemoteSchemaNameQuery name) = do
-  rmSchemas <- scRemoteSchemas <$> askSchemaCache
-  void $ onNothing (Map.lookup name rmSchemas) $
-    throw400 NotExists $ "remote schema with name " <> name <<> " does not exist"
+  sc <- askSchemaCache
+  case Map.lookup name (scRemoteSchemas sc) of
+    Nothing -> if name `elem` getInconsistentRemoteSchemas (scInconsistentObjs sc) then
+                 throw400 NotSupported $
+                 "remote schema with name " <> name <<> " is in inconsistent state; cannot reload"
+               else throw400 NotExists $
+                    "remote schema with name " <> name <<> " does not exist"
+    Just _ -> pure ()
```

Comment:
Is it necessary to fail if the remote schema is inconsistent? That seems unnecessary, and it’s certainly unhelpful—maybe someone is trying to reload the remote schema precisely *because* it’s inconsistent! So I’m wondering if we can go with a simpler check like this:

```suggestion
  remoteSchemas <- getAllRemoteSchemas <$> askSchemaCache
  unless (name `elem` remoteSchemas) $ throw400 NotExists $
    "remote schema with name " <> name <<> " does not exist"
```

Would that cause any problems?

Reply:
Console won't show up a remote schema in the UI if it is present in inconsistent metadata. 
> maybe someone is trying to reload the remote schema precisely because it’s inconsistent!

The Idea is the `reload_remote_schema` API should only refresh the consistent remote schema. Use `reload_metadata` with `reload_remote_schema: true` to refresh all remote schemas along with inconsistent ones. So, the explicit error message is thrown when the remote schema is inconsistent.

Reply:
> Console won't show up a remote schema in the UI if it is present in inconsistent metadata.

We can improve this in the console. 

> That seems unnecessary, and it’s certainly unhelpful—maybe someone is trying to reload the remote schema precisely because it’s inconsistent

I completely agree.

Reply:
> Console won't show up a remote schema in the UI if it is present in inconsistent metadata.

This is something console will be changing soon. It doesnt make sense to hide all inconsistent objects.

Reply:
I'll commit the suggestion.